send-transaction-service/src/send_transaction_service.rs

```diff
@@ -298,9 +308,17 @@ impl SendTransactionService {
                     {
                         // take a lock of retry_transactions and move the batch to the retry set.
                         let mut retry_transactions = retry_transactions.lock().unwrap();
-                        let transactions_to_retry = transactions.len();
+                        let mut transactions_to_retry: usize = 0;
```

Comment:
`transactions_to_retry` changed the semantics: now it is number of transactions that haven't reached retry limit.